code/modules/mob/living/bloodcrawl.dm

```diff
@@ -8,14 +8,10 @@
 	invisibility = 60
 	burn_state = LAVA_PROOF
 
-
-obj/effect/dummy/slaughter/relaymove(mob/user, direction)
-	if (!src.canmove || !direction) return
+/obj/effect/dummy/slaughter/relaymove(mob/user, direction)
 	var/turf/newLoc = get_step(src,direction)
-	loc = newLoc
-	src.canmove = 0
-	spawn(1)
-		src.canmove = 1
+	if(newLoc)
+		loc = newLoc
```

Comment:
I am not sure if it is supposed to be calling Crossed(). Mechanically, it's just a way to search for blood to come out of, not a tile by tile movement solution.